### ISSUE TITLE
Dump N-Quads produces (NULL) literal, which is invalid syntax

### ISSUE THREAD

**Issue by jakubklimek** [openlink/virtuoso-opensource#256]
using the dump n-quads function, I got a dump that has invalid syntax, specifically, there is a line
<code>&lt;http://linked.opendata.cz/resource/domain/coi.cz/check-action/221201060254501/postal-address&gt;  &lt;http://schema.org/streetAddress&gt;       (NULL)  &lt;http://linked.opendata.cz/resource/dataset/coi.cz/kontroly&gt; .</code> which is clearly invalid.

This is from ca81fb6d6825120272fd8c9c4538663edd7fe771

**Comment by jakubklimek** [comment]
Still happening in 0b74378067a7fdda16e0f0789d12f80b1028c483

**Comment by jakubklimek** [comment]
Also, when trying to describe the subject, it says:

<pre><code>Virtuoso 22023 Error SR580: RDF box refers to row with RO_ID = 8704480 of table DB.DBA.RDF_OBJ, but no such row in the table
SPARQL query:
define sql:big-data-const 0 
#output-format:text/plain
define sql:describe-mode "LOD"  DESCRIBE <http://linked.opendata.cz/resource/domain/coi.cz/check-action/221201060254501/postal-address></code></pre>

**Comment by HughWilliams** [comment]
When try to use describe which gives and error is this being run on the  source machine the dump is made from or the destination the dumps are reloaded into ?

Are you able to provide a dataset or the source database so we can attempt to reproduce ?

**Comment by jakubklimek** [comment]
This was run on the source machine, because the target machine fails to load the dump due to the syntax error,

Unfortunately, I had to delete the entity with this object so that I could get a valid dump, so I don't have the faulty database anymore.

According to the error, this should be reproducible by manually deleteing a row from the DB.DBA.RDF_OBJ table? This should cause the (NULL) value in turtle/nquads.

Another issue is how the missing object happened, but that I don't know.

**Comment by HughWilliams** [comment]
@jakubklimek: I note you solved this SR580 , in git issue https://github.com/openlink/virtuoso-opensource/issues/120 ... can the same not be applied here ?

**Comment by jakubklimek** [comment]
Yes, I can fix the data, but the issue is that the RDF serializer produces invalid syntax when it encounters this situation. It should not produce the triple at all or maybe produce a warning

**Comment by HughWilliams** [comment]
@jakubklimek  Ok, can you provide the original sample dataset for the http://linked.opendata.cz/resource/dataset/coi.cz/kontroly graph that can be used to recreate and we can  try it locally.

Or would do any of the graphs we have from the last DB you provided exhibit the problem ?

SQL> sparql select distinct ?g where {GRAPH ?g{?s a ?t}};
g
VARCHAR

---

http://www.openlinksw.com/schemas/virtrdf#
http://www.w3.org/2002/07/owl#
http://localhost:8890/sparql
http://linked.opendata.cz/resource/dataset/mfcr/rozvaha
http://linked.opendata.cz/resource/dataset/mfcr/rozvaha/sample
http://linked.opendata.cz/resource/dataset/mfcr/pozvk
http://linked.opendata.cz/resource/dataset/mfcr/vykzz
http://linked.opendata.cz/resource/dataset/mfcr/ciselniky
http://linked.opendata.cz/resource/dataset/mfcr/ppt/2010
http://linked.opendata.cz/resource/dataset/mfcr/ppt
http://linked.opendata.cz/resource/dataset/mfcr/monitor/dsd/ppt
http://linked.opendata.cz/resource/dataset/mfcr/monitor/dsd/vykzz
http://linked.opendata.cz/resource/dataset/mfcr/monitor/dsd/priloha
b3sifp
http://www.openlinksw.com/schemas/oplweb#
virtrdf-label
http://linked.opendata.cz/resource/dataset/mfcr/vykzz/metadata
http://linked.opendata.cz/resource/dataset/mfcr/monitor/dsd/pozvk
http://linked.opendata.cz/resource/dataset/mfcr/monitor/dsd/rozvaha
http://linked.opendata.cz/resource/dataset/mfcr/ppt/metadata
http://linked.opendata.cz/resource/dataset/mfcr/pozvk/metadata
http://linked.opendata.cz/resource/dataset/mfcr/rozvaha/metadata

22 Rows. -- 249 msec.
SQL>
